xlators/mgmt/glusterd/src/glusterd-pmap.c

```diff
@@ -98,16 +98,23 @@ pmap_port_alloc(xlator_t *this)
 {
     struct pmap_registry *pmap = NULL;
     int p = 0;
+    int i;
 
     GF_ASSERT(this);
 
     pmap = pmap_registry_get(this);
 
-    while (true) {
-        /* coverity[DC.WEAK_CRYPTO] */
-        p = (rand() % (pmap->max_port - pmap->base_port + 1)) + pmap->base_port;
-        if (pmap_port_isfree(p)) {
-            break;
+    /* coverity[DC.WEAK_CRYPTO] */
+    p = (rand() % (pmap->max_port - pmap->base_port + 1)) + pmap->base_port;
+    if (pmap_port_isfree(p)) {
+        return p;
+    } else {
+        p = 0;
+        for (i = pmap->base_port; i < pmap->max_port + 1; i++) {
```

Comment:
This makes port assignment less random. If the first attempt returns a used port, then all future attempts will favor the first port of the list.

Using this approach makes it more random:

```c
    p = rand() % (pmap->max_port - pmap->base_port + 1) + pmap_base_port;
    for (i = pmap->base_port; i <= pmap->max_port; i++) {
        if (pmap_port_is_free(p)) {
            return p;
        }
        if (p++ >= pmap->max_port) {
            p = pmap->base_port;
        }
    }
    /* If we get here, it means all ports are in use. */
```

Reply:
Yes, this makes more sense.